src/analysis/asymptotics.lean

```diff
@@ -789,11 +796,32 @@ have eq₃ : is_O f (λ x, f x / g x * g x) l,
   end,
 eq₃.trans_is_o eq₂
 
-theorem is_o_iff_tendsto {f g : α → β} {l : filter α}
+theorem is_o_iff_tendsto {f g : α → 𝕜} {l : filter α}
     (hgf : ∀ x, g x = 0 → f x = 0) :
   is_o f g l ↔ tendsto (λ x, f x / (g x)) l (𝓝 0) :=
 iff.intro tendsto_nhds_zero_of_is_o (is_o_of_tendsto hgf)
 
+theorem is_o_pow_pow {m n : ℕ} (h : m < n) :
+  is_o (λ(x : 𝕜), x^n) (λx, x^m) (𝓝 0) :=
+begin
+  let p := n - m,
+  have p_pos : 0 < p := nat.sub_pos_of_lt h,
+  have : n = m + p := (nat.add_sub_cancel' (le_of_lt h)).symm,
+  simp [this, pow_add],
+  have : (λ(x : 𝕜), x^m) = (λx, x^m * 1), by simp,
+  rw this,
+  apply is_o_mul_left (is_O_refl _ _) _,
+  rw is_o_iff_tendsto,
+  { simp only [div_one],
+    convert (continuous_pow p).tendsto (0 : 𝕜),
+    exact (zero_pow p_pos).symm },
+  { simp }
+end
+
+theorem is_o_pow_id {n : ℕ} (h : 1 < n) :
+  is_o (λ(x : 𝕜), x^n) (λx, x) (𝓝 0) :=
```

Comment:
Here and elsewhere in this PR, is there any reason to write `(λx, x)` instead of `id`?

Reply:
Here it's about comparing `x^n` and `x`, so I thought it was more readable to have both explicitly spelled out. I can change if you want.

Reply:
I really don't mind. I was curious.